pythainlp/translate/core.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+import os
+import tarfile
+from collections import defaultdict
+from functools import partial
+
+from pythainlp.corpus import download, get_corpus_path
+from pythainlp.tools import get_full_data_path, get_pythainlp_data_path
+from fairseq.models.transformer import TransformerModel
+from sacremoses import MosesTokenizer, MosesDetokenizer
+from pythainlp.tools import get_full_data_path, get_pythainlp_data_path
+from pythainlp.tokenize import word_tokenize as th_word_tokenize
+
+en_word_detokenize = MosesDetokenizer("en")
+en_word_tokenize = MosesTokenizer("en")
+th_word_tokenize = partial(th_word_tokenize, keep_whitespace=False)
+
+
+def download_model() -> None:
+    """
+    Download Model
+    """
+    if get_corpus_path("scb_1m_th-en_newmm") is None:
+        download("scb_1m_th-en_newmm", force=True, version="1.0")
+        tar = tarfile.open(get_corpus_path("scb_1m_th-en_newmm"), "r:gz")
+        tar.extractall()
+        tar.close()
+    if get_corpus_path("scb_1m_th-en_spm") is None:
+        download("scb_1m_th-en_spm", force=True, version="1.0")
+        tar = tarfile.open(get_corpus_path("scb_1m_th-en_spm"), "r:gz")
+        tar.extractall()
+        tar.close()
+    if get_corpus_path("scb_1m_en-th_moses") is None:
+        download("scb_1m_en-th_moses", force=True, version="1.0")
+        tar = tarfile.open(get_corpus_path("scb_1m_en-th_moses"), "r:gz")
+        tar.extractall()
+        tar.close()
```

Comment:
So, we download all the models for the first run? What do you think we just download what you're trying to use?

For example, if I plan to use only `scb_1m_th-en_newmm`; I don't need to download the other three models.

Reply:
@heytitle  OK. I fixed it.